src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -64,26 +62,14 @@ public DateTime getTimestamp()
   @Override
   public List<String> getDimension(String dimension)
   {
-    Object dimValue = event.get(dimension);
+    final Object dimValue = event.get(dimension);
 
     if (dimValue == null) {
-      return Lists.newArrayList();
+      return Collections.emptyList();
     } else if (dimValue instanceof List) {
-      return Lists.transform(
-          (List) dimValue,
-          new Function<Object, String>()
-          {
-            @Override
-            public String apply(Object input)
-            {
-              return String.valueOf(input);
-            }
-          }
-      );
-    } else if (dimValue instanceof Object) {
-      return Arrays.asList(String.valueOf(event.get(dimension)));
+      return Lists.transform((List<Object>) dimValue, TO_STRING_INCLUDING_NULL);
```

Comment:
personally, I preferred the function definition being here itself specially bcoz  this is the only place TO_STRING_INCLUDING_NULL is being used. You could add the comment for why it was chosen instead of guava's.